add examples to Interactive API Cloud UI Docs (#2122)
PrefectHQ/prefect: docs/cloud/ui/interactive-api.md

```graphql
query {
  flow (where: {name: {_eq: "My Flow Name"}}){
    name 
    id
    tasks {
      name
    }
  }
}
```

We can also filter the results by looking only for flows with tasks with a certain name:

```graphql
query {
  flow (where: {tasks: {name:{_eq: "x"}}})
     {
      name
      id
      tasks {
        name
      }
    }
}
```

And then limit the results further to only show the tasks that have that certain name:

```graphql
query {
  flow (where: {tasks: {name:{_eq: "x"}}})
     {
      name
      id
      tasks (where: {name: {_eq: "x"}}) {
        name
      }
    }
}
```
You can see more about query filters in the [Hasura Docs](https://hasura.io/docs/1.0/graphql/manual/queries/query-filters.html).

### Schema

In the docs for the Interactive API (included on the Interactive API page itself) you can find the schema - information about the queries and mutations you can run and more information about what fields you can request (for queries) or change (for mutations).

![](/cloud/ui/interactive-api-inline-docs.png)  

Finally, as you read further through the Prefect Cloud docs, look out for the GraphQL header:
### GraphQL <Badge text="GQL"/>

> This header shows that we are giving an example of a query or mutation you can run using the Interactive API. 

Reviewer: Could we add a heading to this section, something like `More Examples` maybe? Also the Y-margin around `### GraphQL` breaks up the section pretty oddly, maybe this could be a screenshot instead?

Author: I tried out a screenshot but it didn't section much better.  Switched to referencing the badge instead.

Author: @znicholasbrown - ready for another review!

# Interactive API

The Interactive API is an embedded GraphQL client that allows you to access all of your data with complete flexibility. Thanks to features like schema introspection and automatic authentication, the interactive API makes it simple to access Cloud. You can also see the GraphQL API documentation directly inline.  

![](/cloud/ui/interactive-api.png)

> ### Example Query

Reviewer: ```suggestion
### Basic Query
```

> To find out information about your flows, you could use a flow query.  The basics for that query would be:

Reviewer: We should add some sort of transition here, something like:

```suggestion
Let's start with an example of how you might find basic information about your flows.
```

```graphql
query {
    flow {
        name
        id
    }
}
```

> You can also use the autocomplete to show what other fields you could query:

Reviewer: ```suggestion
Notice that when we want to get the `flow` object, we must wrap it in a `query` block. This tells GraphQL that we're _retrieving_ something from the schema, as opposed to a `mutation` block which allows us to manipulate our data.

We've told the server to return a list of `flow` objects, each containing their respective `name` and `id` fields. As you interact with the schema in the editor window, you'll see an autocomplete window that will show hints of other available fields, which you can select to insert into your query. 
```

Reviewer: I think we should provide more instruction of what's going on here.

![](/cloud/ui/dropdown-on-interactive-api.png)

### Nested Query

> To get information about the tasks connected to your flows, you could use a nested query:

Reviewer: ```suggestion
Let's build on our earlier query to get information about the tasks associated with our flows. One way to accomplish this would be to use a nested query:
```

```graphql
query {
  flow {
    name 
    id
    tasks {
      name
    }
  }
}
```

> 

Reviewer: ```suggestion
We've nested `tasks` within the `flow` object, which tells the server to retrieve tasks only within the context of each flow that it returns.
```

### Limits and Offset

To limit the number of items that are returned, you can use the Limit selector at the top of the Interactive API page. The default limit is 10 and the maximum is 100. Inline limit arguments are overriden by the value set in the Limit selector. 

> To the left of the Limit selector there is also an Offset selector. The Offset selector lets you say where your queries should start. So for example, if your query returned 5 flows and you set the limit to 2 and the offset to 0, you would see the first 2 items returned from your query.  If you set the limit to 2 and the offset to 2, you would see the next 2 items. 

Reviewer: ```suggestion
To the left of the Limit selector there is also an Offset selector. The Offset selector tells the server at which index your queries should start. For example, if your unlimited query would return 5 flows and you set the limit to 2, an offset of 0 would would return the first two items in the set. To get the next two items with no overlap, you would set the offset to 2.
```

### Query Filters

> Another way to limit the results returned would be to filter your search results.  For the flow query example, you could filter the results by looking only for flows with a certain name using the 'where' argument:

Reviewer: ```suggestion
Another way to limit the results returned would be to filter your search results. Building on our earlier flow query, we add a `where` argument to look for flows with a certain name:
```

```graphql
query {
  flow (where: {name: {_eq: "My Flow Name"}}){
    name 
    id
    tasks {
      name
    }
  }
}
```

> You could also filter the results by looking only for flows with tasks with a certain name:

Reviewer: ```suggestion
We can also filter the results by looking only for flows with tasks with a certain name:
```

```graphql
query {
  flow (where: {tasks: {name:{_eq: "x"}}})
     {
      name
      id
      tasks {
        name
      }
    }
}
```

And then limit the results further to only show the tasks that have that certain name:

```graphql
query {
  flow (where: {tasks: {name:{_eq: "x"}}})
     {
      name
      id
      tasks (where: {name: {_eq: "x"}}) {
        name
      }
    }
}
```

> ### Docs

Reviewer: I wonder if we might move the links to the Hasura and GraphQL docs into the contexts above, like a `Read more about query filters in the [Hasura Docs](https://hasura.io/docs/1.0/graphql/manual/queries/query-filters.html)` in each section.

Author: Good suggestion - I've updated.

![](/cloud/ui/interactive-api.png)

### Basic Query

Let's start with an example of how you might find basic information about your flows.

```graphql
query {
    flow {
        name
        id
    }
}
```
Notice that when we want to get the `flow` object, we must wrap it in a `query` block. This tells GraphQL that we're _retrieving_ something from the schema, as opposed to a `mutation` block which allows us to manipulate our data.

We've told the server to return a list of `flow` objects, each containing their respective `name` and `id` fields. As you interact with the schema in the editor window, you'll see an autocomplete window that will show hints of other available fields, which you can select to insert into your query. 

![](/cloud/ui/dropdown-on-interactive-api.png)

You can see more information about GraphQL mutations and queries in the [GraphQL Docs](https://graphql.org/learn/)

### Nested Query

Let's build on our earlier query to get information about the tasks associated with our flows. One way to accomplish this would be to use a nested query:

```graphql
query {
  flow {
    name 
    id
    tasks {
      name
    }
  }
}
```

> We've nested `tasks` within the `flow` object, which tells the server to retrieve tasks only within the context of each flow that it returns.

Reviewer 2: ```suggestion
```

We've nested `tasks` within the `flow` object, which tells the server to retrieve tasks only within the context of each flow that it returns.
```

Reviewer 2: This actually does not affect rendering

> ### Limits and Offset

Reviewer 2: ```suggestion
We've nested `tasks` within the `flow` object, which tells the server to retrieve tasks only within the context of each flow that it returns.

### Limits and Offset
```

Reviewer 2: This actually does not affect rendering
